### PR TITLE
add related posts on single post page

### PR DESCRIPTION
**Pull request by dusmel**
#### What does this PR do?
This PR adds related posts when viewing a single post

#### Description of Task to be completed?
• handle when no post is related
• remove the post itself to be repeated on the related ones
• redesign single post page to make place for the new block

#### How should this be manually tested?
• checkout to this branch `ft-related-posts`
• Click on one of the posts
• you will notice a new block on the right side displaying related posts

#### Screenshots (if appropriate)
# 1
image
# 2
image

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "plugins": ["react"],
   "rules": {
     "no-param-reassign": 0,
+    "no-console": [1, { "allow": ["error"] }],
```
Comment (gracelungu) [inline .eslintrc.json:23]: @dusmel Why did you add this rule?

File: src/actions/fetchSinglePostAction.js [hunk 2]
```diff
@@ -36,6 +50,7 @@ export default (slug) => async (dispatch) => {
 
     dispatch(fetchSuccess(post));
     dispatch(toggleLoading(false));
+    dispatch(await fetchRelatedPosts(post.type, slug));
```
Comment (gracelungu) [inline src/actions/fetchSinglePostAction.js:63]: @dusmel Why are you awaiting this action?

File: src/pages/SinglePost/SinglePost.jsx [hunk 3]
```diff
@@ -6,11 +6,7 @@ import Post from '../../components/SinglePost/SinglePost';
 const SinglePost = ({ match, location, history }) => (
   <Layout match={match} location={location} history={history}>
     <div className="container-fluid">
-      <div className="row">
-        <div className="col-md-8 offset-md-2">
-          <Post slug={match.params.slug} />
-        </div>
-      </div>
```
Comment (gracelungu) [inline src/pages/SinglePost/SinglePost.jsx:13]: @dusmel I do not understand why you deleted this!

File: src/actions/fetchSinglePostAction.js [hunk 4]
```diff
@@ -20,6 +21,19 @@ export const fetchSuccess = (post) => ({
   payload: post,
 });
 
+const fetchRelatedPosts = (category, singlePostSlug) => server.get(`feed?offset=0&limit=3&category=${category}`)
```
Comment (gracelungu) [inline src/actions/fetchSinglePostAction.js]: @dusmel I think we should create a constant to hold the limit for `related posts` and increase this one a little bit more.

File: src/store/initialState.js [hunk 5]
```diff
@@ -22,4 +22,18 @@ module.exports = {
...
+      userType: '',
+      description: '',
+      likesCount: 0,
+      createdAt: '',
+    },
+    relatedPosts: {
+      isEmpty: false,
+      feed: [],
+    },
+  },
```
Comment (gracelungu) [inline src/store/initialState.js:38]: @dusmel Why did you add this?

**Comment by gracelungu** [comment]
@dusmel What if we just get the latest posts once no related posts are found. And in this case, we will have to remove the title!

File: src/store/initialState.js (hunk 5 above)
Comment (dusmel) [reply src/store/initialState.js:38]: Because someone was using them without giving them a default value ( and made them required). it was complaining.

File: src/pages/SinglePost/SinglePost.jsx (hunk 3 above)
Comment (dusmel) [reply src/pages/SinglePost/SinglePost.jsx:13]: Because I couldn't add the new block in single post page, since the Post component only had `col-md-8`, I had to give it a full width then divide in the Post component.

File: .eslintrc.json (hunk 1 above)
Comment (dusmel) [reply .eslintrc.json:23]: Because right now we don't have a way of handling errors and it was giving me a hard time to debug so I added a rule to allow only console.error to know why codes aren't behaving the way they are supposed to!

File: src/actions/fetchSinglePostAction.js (hunk 2 above)
Comment (dusmel) [reply src/actions/fetchSinglePostAction.js:63]: because otherwise, it will return a promise and dispatch take an object

File: src/actions/fetchSinglePostAction.js (hunk 4 above)
Comment (dusmel) [reply src/actions/fetchSinglePostAction.js]: So the post itself can't go beyond 55 characters that mean is very short, adding more cards on the right side will make the whole design looks weird

**Comment by dusmel** [comment]
@gracelungu This problem will be temporary because I am sure just after deployment it will be rare to find a whole section with posts less than 2. Also, it will make this feature complex because its only dealing with related posts

**Comment by dusmel** [comment]
Also @gracelungu can you be ASK with your comments, why are you Mobutung on my PR manze?

**Review by gracelungu — APPROVED** [review]

**Review by Karlmusingo — APPROVED** [review]

File: src/actions/fetchSinglePostAction.js [hunk 6]
```diff
@@ -20,6 +21,19 @@ export const fetchSuccess = (post) => ({
...
+  .then(({ data: { feed } }) => {
+    const filteredFeed = feed.filter(({ slug }) => slug !== singlePostSlug);
+    return {
+      type: types.FETCH_RELATED_POSTS,
+      payload: {
+        isEmpty: !filteredFeed.length,
+        feed: filteredFeed,
+      },
```
Comment (gracelungu) [inline src/actions/fetchSinglePostAction.js]: @dusmel Would you mind creating a function outside to return this object?

File: src/actions/fetchSinglePostAction.js [hunk 7]
```diff
@@ -20,6 +21,19 @@ export const fetchSuccess = (post) => ({
   payload: post,
 });
 
+const fetchRelatedPosts = (category, singlePostSlug) => server.get(`feed?offset=0&limit=3&category=${category}`)
+  .then(({ data: { feed } }) => {
```
Comment (gracelungu) [inline src/actions/fetchSinglePostAction.js]: @dusmel Would you mind using the pattern found in the codebase?

**Review by gracelungu — CHANGES_REQUESTED** [review]

**Comment by gracelungu** [comment]
[Add related posts while viewing a single post](https://trello.com/c/tSkeW8qb/78-add-related-posts-while-viewing-a-single-post)
